### ISSUE TITLE
`Dataset.spei(_normalized)` = [704], so why `Plotter._calculateDenormalizedValues()` only gets [342] as input?

### ISSUE THREAD

**Issue by A-Infor** [A-Infor/Python-OOP-LSTM-Drought-Predictor#43]
Inside `Plotter._calculateDenormalizedValues()`, `len(true_values_normalized_dict['100%'].flatten())` = 342.
Shouldn't 100% be like, 100%? That is, 704?

**Comment by A-Infor** [comment]
Further investigation reveals that, when `NeuralNetwork.use_neural_network()` calls `Plotter.plotModelPlots()`, it passes:

1. `dataTrueValues_dict['100%']` (normalized) with dimension (57,6);
2. (`predictValues_dict['80%']` + `predictValues_dict['20%']`) (normalized) with the same dimension.

So the answer to que question is upstream.

But it can be noticed that `predictValues_dict` is built inside `NeuralNetwork.use_neural_network()`, and that `dataTrueValues_dict` is returned inside this same method, from a call to `Dataset.format_data_for_model()`.

**Comment by A-Infor** [comment]
So, let's see...

1. `NeuralNetwork.use_neural_network()` calls `Dataset.format_data_for_model()` passing nothing relevant for the question;
2. `Dataset.format_data_for_model()` calls `Dataset._train_test_split()`, then `Dataset._create_input_output_pairs()`;
3. the first method returns portions for 20%, 80%, and 100%;
4. the second one splits the portions into input and output, thus halving them to (57,6) = 342.

The outputs are:

1. `spei_dict`;
2. `months_dict`;
3. `dataForPrediction_dict`;
4. `dataTrueValues_dict`;
5. `monthsForPrediction_dict`;
6. `monthsForPredicted_dict`.

The first two are simply the output of `Dataset._train_test_split()`.
The others are the output of `Dataset._create_input_output_pairs()`.

**Comment by A-Infor** [comment]
Following the trail, `Plotter.showPredictionsDistribution()` have the following parameters:

1. `spei_provided_inputs`;
2. `spei_predicted_values`.

The arguments have the same names, for then sake of simplicity.

It then calls `Plotter._calculateDenormalizedValues()`, providing the same arguments as parameters.
The names are the same for the sake of simplicity:

1. `spei_provided_inputs`: (57, 6) = 342;
2. `spei_predicted_values['80%']` + `spei_predicted_values['20%']`: (46, 6) + (11, 6) = (57, 6) = 342.

Fine, so there are 342 inputs and 342 outputs. But they are REAL outputs, not expected ones.
Maybe the trick is here.

When checking `np.max()` and `np.min()` values on both datasets, when they are denormalized, they do not match the original `np.max()` and `np.min()` values because they are NOT the same data!

`spei_predicted_values` != `spei_expected_outputs`

The previous naming was confusing and did not allow me to perceive this brefore.
